components/menu/sub_menu.tsx

```diff
@@ -0,0 +1,182 @@
+        event.preventDefault();
+        setAnchorElement(event.currentTarget);
+    };
+
+    const handleSubMenuClose = (event: MouseEvent<HTMLLIElement>) => {
+        event.preventDefault();
+        setAnchorElement(null);
+    };
+
+    const hasSubmenuItems = Boolean(props.children);
+    if (!hasSubmenuItems) {
+        return null;
+    }
+
+    if (isMobileView) {
+        function MenuModalComponent() {
+            function handleModalExited() {
+                dispatch(closeModal(props.menuId));
+            }
+
+            function handleModalClickCapture() {
+                handleModalExited();
+            }
+
+            return (
+                <CompassDesignProvider theme={theme}>
+                    <GenericModal
+                        id={props.menuId}
+                        ariaLabel={props.menuAriaLabel}
+                        onExited={handleModalExited}
+                        backdrop={true}
+                        className='menuModal'
+                    >
+                        <MuiMenuList
+                            aria-labelledby={props.triggerId}
+                            onClick={handleModalClickCapture}
+                        >
+                            {props.children}
+                        </MuiMenuList>
+                    </GenericModal>
+                </CompassDesignProvider>
+            );
+        }
+
+        function handleAnchorButtonClickOnMobile(event: MouseEvent<HTMLLIElement>) {
+            event.preventDefault();
+
+            dispatch(openModal({
+                modalId: props.menuId,
+                dialogType: MenuModalComponent,
+            }));
+        }
+
+        return (
+            <MuiMenuItem
+                id={props.triggerId}
+                disableRipple={true}
+                aria-controls={props.menuId}
+                aria-haspopup='true'
+                onClick={handleAnchorButtonClickOnMobile}
+                sx={{padding: '0'}}
+            >
+                {props.triggerElement}
+            </MuiMenuItem>
+        );
+    }
+
+    return (
+        <MuiMenuItem
+            id={props.triggerId}
+            aria-controls={isSubMenuOpen ? props.menuId : undefined}
+            aria-haspopup='true'
+            aria-expanded={isSubMenuOpen ? 'true' : undefined}
+            aria-label={props.triggerAriaLabel}
+            disableRipple={true}
+            onMouseEnter={handleSubMenuOpen}
+            onMouseLeave={handleSubMenuClose}
+            sx={{padding: '0'}}
+        >
+            {props.triggerElement}
+            <MuiMenuStyled
+                id={props.menuId}
+                anchorEl={anchorElement}
+                open={isSubMenuOpen}
+                aria-label={props.menuAriaLabel}
+                sx={{pointerEvents: 'none'}} // disables the menu background wrapper
+                {...getOriginOfAnchorAndTransform(props.openAt)}
+            >
+                <MuiMenuList
```

Comment:
And same comment here about the extra `MenuList` inside of the `Menu`. Since you're not setting `padding: 0` on this one, you're also ending up with slightly more vertical padding here
![Screen Shot 2022-12-08 at 12 45 13 PM](https://user-images.githubusercontent.com/3277310/206532271-404c6543-4355-416b-bde2-bef0669e7ee5.png)

Reply:
Removed the extra MenuList